examples/all-clusters-app/all-clusters-common/src/laundry-washer-controls-delegate-impl.cpp

```diff
@@ -54,3 +55,44 @@ CHIP_ERROR LaundryWasherControlDelegate::GetSupportedRinseAtIndex(size_t index,
     supportedRinse = LaundryWasherControlDelegate::supportRinsesOptions[index];
     return CHIP_NO_ERROR;
 }
+
+Status LaundryWasherControlDelegate::PreAttributeCheck(AttributeId attributeId, uint16_t size, uint8_t * value)
+{
+    switch (attributeId)
+    {
+    case Attributes::SpinSpeedCurrent::Id: {
+        if (*value >= ArraySize(spinSpeedsNameOptions))
```

Comment:
Why is this check being done in the delegate?  The cluster could do this check, right?

But also, this check happens to be wrong, which is another reason this check should be done in the cluster, where we have more of a chance of getting the edge cases right....

Reply:
The SpinSpeedCurrent should be the index which should be < supported spin-speed list. Since the list may be different, so I  put it in the delegate. 
I will move the checking to the cluster and get the size from the delegate.